eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Rules/TestChain.hs

```diff
@@ -160,7 +154,45 @@ import Test.QuickCheck
     (.||.),
     (===),
   )
+import Test.Tasty (TestTree, defaultMain, testGroup)
+import qualified Test.Tasty.QuickCheck as TQC
 
+-- ==========================================
+
+test1 ::
+  forall era.
+  ( EraGen era,
+    State (EraRule "PPUP" era) ~ PPUPState era,
+    QC.HasTrace (CHAIN era) (GenEnv era)
+  ) =>
+  Property
+test1 =
+  forAllChainTrace @era longTraceLen $ \tr -> do
+    let ssts :: [SourceSignalTarget (CHAIN era)]
+        -- Signal(CHAIN era) = Block (BHeader (EraCrypto era)) era
+        ssts = sourceSignalTargets tr
+        -- noEpochBoundarySsts = filter sameEpoch ssts
+        justBoundarySsts = filter (not . sameEpoch) ssts
+
+    conjoin (map (checkPreservation @era) (zip justBoundarySsts [0 ..]))
+
+minimal ::
+  forall era.
+  ( EraGen era,
+    QC.HasTrace (CHAIN era) (GenEnv era),
+    State (EraRule "PPUP" era) ~ PPUPState era
+  ) =>
+  TestTree
+minimal =
+  testGroup
+    "Minimal Property Tests"
+    [ TQC.testProperty "total amount of Ada is preserved (Chain)" (test1 @era)
+    ]
+
+main :: IO ()
+main = defaultMain (minimal @(ShelleyEra TestCrypto))
+
+-- :main --quickcheck-replay=443873
```

Comment:
a bug we've now squashed?

Reply:
Yes. sometimes it is nice to leave some infrastructure in place.
I moved it to a comment.